cmd/juju/model/cancelgeneration.go

```diff
@@ -94,10 +95,15 @@ func (c *cancelGenerationCommand) Run(ctx *cmd.Context) error {
 	}
 	defer client.Close()
 
-	// TODO (hml) 20-12-2018
-	// update to check err when CancelGeneration() is implemented in the
-	// apiserver.
-	client.CancelGeneration()
+	_, modelDetails, err := c.ModelCommandBase.ModelDetails()
+	if err != nil {
+		return errors.Annotate(err, "getting model details")
+	}
+
+	modelTag := names.NewModelTag(modelDetails.ModelUUID)
+	if err = client.CancelGeneration(modelTag); err != nil {
+		return err
+	}
 
 	ctx.Stdout.Write([]byte("changes dropped and target generation set to current\n"))
```

Comment:
I don't think we should refer to the changes as "dropped" as the next generation changes will be applied (according to the spec). Perhaps there should be a y/n confirmation with this detail.

Reply:
We should also give some feedback when a generation is auto-completed, but that can come in a future patch.

Reply:
Agreed, dropped came from the spec.  @mitechie: preferences with rewording?  @manadart: https://trello.com/c/KPexZiQJ/998-add-succeeded-message-to-advance-command-when-auto-complete-happens

Reply:
"remaining incomplete changes dropped and... "